src/Evaluator.ml

```diff
@@ -82,6 +133,7 @@ let rec eval_open gamma expr =
           eval_open (Env.extend_l c_gamma idl argv) body
       | _ -> Result.fail "Trying to apply an argument to a non-function" )
   | Box e -> return @@ Val.Box e
+  | Let (idr, bounded_e, body) -> eval_open gamma (subst_r bounded_e idr body)
```

Comment:
Here one does not explicit substitution. You just need to evaluate the bound expression, add a new binding to the regular context and evaluate the body in this updated context.